File: accounts/accounts-domain/src/main/java/com/hrrm/famoney/accounts/movement/Movement.java
package com.hrrm.famoney.accounts.movement;

import com.hrrm.famoney.accounts.Account;
import com.hrrm.famoney.accounts.AccountsDomainEntity;
import lombok.*;
import lombok.experimental.Accessors;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@Table(
        schema = AccountsDomainEntity.SCHEMA_NAME,
        name = "movement"
)
@DiscriminatorColumn(name = "type")
@Accessors(chain = true)
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(
        onlyExplicitlyIncluded = true,
        callSuper = true
)
public abstract class Movement extends AccountsDomainEntity {

    public static final String FIND_MOVEMENTS_WITH_PAGINATION = "com.hrrm.famoney.domain.accounts.movement.AccountMovement#finMovenetsWithPagination";
    public static final String ACCOUNT_ID_PARAMETER_NAME = "accountId";
    public static final String MOVEMENT_DATE_PARAMETER_NAME = "movementDateId";

    @ManyToOne
    @JoinColumn(name = "account_id")
    @NotNull
    private Account account;

    @Column(name = "date")
    @NotNull
    private LocalDate date;

    @Column(name = "pos")
    @NotNull
    private Integer position;

    @Column(name = "booking_date")
    @Nullable
    private LocalDate bookingDate;

    @Column(name = "budget_period")
    @Nullable
    private LocalDate budgetPeriod;

    @Column(name = "amount")
    @Nullable
    private BigDecimal amount;

    @Column(name = "total")
    @Nullable
    private BigDecimal total;

}

File: accounts/accounts-domain/src/main/kotlin/com/hrrm/famoney/accounts/movement/Movement.kt
package com.hrrm.famoney.accounts.movement

import com.hrrm.famoney.accounts.Account
import com.hrrm.famoney.accounts.AccountsDomainEntity
import java.math.BigDecimal
import java.time.LocalDate
import javax.persistence.*

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@Table(schema = AccountsDomainEntity.SCHEMA_NAME, name = "movement")
@DiscriminatorColumn(name = "type")
abstract class Movement : AccountsDomainEntity() {
    @ManyToOne
    @JoinColumn(name = "account_id")
    lateinit var account: Account

    @Column(name = "date")
    lateinit var date: LocalDate

    @Column(name = "pos")
    var position: Int = 0

    @Column(name = "booking_date")
    var bookingDate: LocalDate? = null

    @Column(name = "budget_period")
    var budgetPeriod: LocalDate? = null

    @Column(name = "amount")
    var amount: BigDecimal = BigDecimal.ZERO

    @Column(name = "total")
    var total: BigDecimal = BigDecimal.ZERO

    companion object {
        const val FIND_MOVEMENTS_WITH_PAGINATION =
            "com.hrrm.famoney.domain.accounts.movement.AccountMovement#finMovenetsWithPagination"
        const val ACCOUNT_ID_PARAMETER_NAME = "accountId"
        const val MOVEMENT_DATE_PARAMETER_NAME = "movementDateId"
    }
}